src/transformers/pipelines/__init__.py

```diff
@@ -859,6 +859,9 @@ def pipeline(
                 'You cannot use both `pipeline(... torch_dtype=..., model_kwargs={"torch_dtype":...})` as those'
                 " arguments might conflict, use only one.)"
             )
+        if torch_dtype is str:
+            if hasattr(torch, torch_dtype):
```

Comment:
if `torch_dtype` is an attribute of torch, then it will do the conversion successfully otherwise it will pass the str as is. The otherwise scenario handles strings such as "auto" as specified in the function docstring.